src/stdlib_strings_format_string.fypp

```diff
@@ -0,0 +1,52 @@
+#:include "common.fypp"
+#:set RIL_KINDS_TYPES = REAL_KINDS_TYPES + INT_KINDS_TYPES + LOG_KINDS_TYPES
+submodule (stdlib_strings) stdlib_strings_format_string
+
+    implicit none
+    integer, parameter :: buffer_len = 512
+
+contains
+
+    #:for kind, type in RIL_KINDS_TYPES
+    module procedure format_string_${type[0]}$${kind}$
+        !! Format ${type}$ variable as character sequence
+        character(len=buffer_len) :: buffer
+        integer :: stat
+        if(present(fmt)) then
+            write(buffer, fmt, iostat=stat) val
+            if(stat == 0) then
+                string = trim(adjustl(buffer))
+            else
+                string = '*'
+                    !!\TODO: *?
```

Comment:
For a meaningful number of asterisks in the output we must be able to parse format strings and determine the width of the field we are writing into. Just writing a single asterisk until we are able to do so in stdlib sounds like a good compromise for now. What do you think?

Reply:
![image](https://user-images.githubusercontent.com/32035096/124059840-599a5680-da5e-11eb-9410-0400ab652984.png)
I think the `*` sign here should be distinguished a little bit more, for example, written as `[*]`, `!*!` and the like can be distinguished from the `*****` sign (too narrow formatter for variables) **before and after it**.

Reply:
Checking a failed formatter should be simple, currently I'm using `starts_with(format_string(var, "(...)"), "*")`. This is not fail-safe as it would match a formatter like `'("*", g0)'` as well. Using `format_string` should imply that we trade some of the versatility of the internal IO approach for having a function.

How about returning an empty string on failure? This would be clearly distinct from a too narrow formatter?

Reply:
It is mainly to remind **the users**: an empty string may not be effectively fed back to the user, this is a `format_string` failure.

1. We need to notify the user **more clearly that this has failed**;
2. The string output after failure is **simple enough**;
3. **Does not hinder the effective operation** of the main process of our `Fortran` program.